runtime/context.go

```diff
@@ -124,6 +210,19 @@ func accumulateLogFields(ctx context.Context, newFields []zap.Field) []zap.Field
 	return fields
 }
 
+// ExtractScope extracts scope fields from a context into a tag.
+func ExtractScope(ctx context.Context, key string) map[string]string {
+	tags := map[string]string{}
+	if fn, ok := ContextScopeExtractors[key]; ok {
```

Comment:
I thought this would be part of `Gateway` struct like most of the other singletons, that way we could have multiple per application or concurrently.